src/Dfe.Spi.EventBroker.Application/Send/SendManager.cs

```diff
@@ -0,0 +1,90 @@
+            ISubscriptionRepository subscriptionRepository,
+            IRestClient restClient,
+            ILoggerWrapper logger)
+        {
+            _distributionRepository = distributionRepository;
+            _eventRepository = eventRepository;
+            _subscriptionRepository = subscriptionRepository;
+            _restClient = restClient;
+            _logger = logger;
+        }
+        
+        public async Task SendAsync(Distribution distribution, CancellationToken cancellationToken)
+        {
+            _logger.Debug($"Reading latest distribution data for {distribution.Id} in subscription {distribution.SubscriptionId}");
+            distribution = await _distributionRepository.GetAsync(distribution.Id, distribution.SubscriptionId, cancellationToken);
+            if (distribution.Status == DistributionStatus.Sent)
+            {
+                _logger.Info($"Stopping processing distribution {distribution.Id} in subscription {distribution.SubscriptionId} as it is already sent");
+                return;
+            }
+            
+            _logger.Debug($"Reading event {distribution.EventId}");
+            var @event = await _eventRepository.GetAsync(distribution.EventId, cancellationToken);
+            
+            _logger.Debug($"Reading subscription {distribution.SubscriptionId} for {@event.Publisher}.{@event.EventType}");
+            var subscription = await _subscriptionRepository.GetSubscriptionToEventAsync(@event.Publisher, @event.EventType,
+                distribution.SubscriptionId, cancellationToken);
+
+            distribution.Attempts++;
+            try
+            {
+                await SendToSubscriberAsync(@event, subscription, cancellationToken);
+                _logger.Info($"Sent event {@event.Id} to subscriber {subscription.Id} at {subscription.EndpointUrl}");
+
+                distribution.Status = DistributionStatus.Sent;
+                await _distributionRepository.UpdateAsync(distribution, cancellationToken);
+            }
+            catch (Exception)
+            {
+                distribution.Status = distribution.Attempts >= 5 
+                    ? DistributionStatus.Failed 
+                    : DistributionStatus.PendingRetry;
+                await _distributionRepository.UpdateAsync(distribution, cancellationToken);
+                
+                throw;
+            }
+        }
+
+        private async Task SendToSubscriberAsync(Event @event, Subscription subscription, CancellationToken cancellationToken)
+        {
+            _logger.Debug($"Sending event {@event.Id} to subscriber {subscription.Id} at {subscription.EndpointUrl}");
+            var request = new RestRequest(subscription.EndpointUrl, Method.POST);
+            request.AddParameter(string.Empty, @event.Payload, "application/json", ParameterType.RequestBody);
+            var response = await _restClient.ExecuteTaskAsync(request, cancellationToken);
```

Comment:
Visual Studio is complaining to me about this:

`Warning	CS0618	'IRestClient.ExecuteTaskAsync(IRestRequest, CancellationToken)' is obsolete: 'Use ExecuteAsync instead'`